rust/lance-index/src/scalar/inverted/wand.rs

```diff
@@ -526,28 +526,25 @@ impl<'a, S: Scorer> Wand<'a, S> {
                     posting
                         .positions(posting.index)
                         .expect("positions must exist"),
-                    posting.position,
+                    posting.position as i32,
                 )
             })
             .collect::<Vec<_>>();
+        position_iters.sort_unstable_by_key(|iter| iter.position_in_query);
 
         loop {
-            let mut max_pos = None;
+            let mut max_relative_pos = None;
             let mut all_same = true;
-            for iter in &position_iters {
-                match (iter.relative_position(), max_pos) {
-                    (Some(pos), None) => {
-                        max_pos = Some(pos);
-                    }
-                    (Some(pos), Some(max)) => {
-                        if pos > max {
-                            max_pos = Some(pos);
-                        }
-                        if pos != max {
-                            all_same = false;
-                        }
-                    }
-                    _ => return false,
+            for window in position_iters.windows(2) {
+                let last = window[0].relative_position();
+                let next = window[1].relative_position();
+                let (Some(last), Some(next)) = (last, next) else {
+                    return false;
+                };
+                max_relative_pos = Some(std::cmp::max(last, next) - slop);
+                if !(last <= next && next <= last + slop) {
```

Comment:
No change needed but I see you have the check `last <= next`.  Is it possible that `last > next`?